Adjust Grass 6.4.1 to work with ffmpeg 8.

2011-11-17 Fabio Erculiani
2011-12-08 Martin von Gagern

References:
https://bugs.gentoo.org/390827
https://bugs.gentoo.org/392371

--- grass-6.4.1.orig/lib/ogsf/gsd_img_mpeg.c
+++ grass-6.4.1/lib/ogsf/gsd_img_mpeg.c
@@ -66,7 +66,7 @@ static AVStream *add_video_stream(AVForm
 
     c = st->codec;
     c->codec_id = codec_id;
-    c->codec_type = CODEC_TYPE_VIDEO;
+    c->codec_type = AVMEDIA_TYPE_VIDEO;
 
     /* put sample parameters */
     c->bit_rate = 400000;
@@ -215,7 +215,7 @@ static void write_video_frame(AVFormatCo
 
 	av_init_packet(&pkt);
 
-	pkt.flags |= PKT_FLAG_KEY;
+	pkt.flags |= AV_PKT_FLAG_KEY;
 	pkt.stream_index = st->index;
 	pkt.data = (uint8_t *) picture;
 	pkt.size = sizeof(AVPicture);
@@ -236,7 +236,7 @@ static void write_video_frame(AVFormatCo
 		av_rescale_q(c->coded_frame->pts, c->time_base,
 			     st->time_base);
 	    if (c->coded_frame->key_frame)
-		pkt.flags |= PKT_FLAG_KEY;
+		pkt.flags |= AV_PKT_FLAG_KEY;
 	    pkt.stream_index = st->index;
 	    pkt.data = video_outbuf;
 	    pkt.size = out_size;
@@ -301,10 +301,10 @@ int gsd_init_mpeg(const char *filename)
     av_register_all();
 
     /* auto detect the output format from the name. default is mpeg. */
-    fmt = guess_format(NULL, filename, NULL);
+    fmt = av_guess_format(NULL, filename, NULL);
     if (!fmt) {
 	G_warning(_("Unable to deduce output format from file extension: using MPEG"));
-	fmt = guess_format("mpeg", NULL, NULL);
+	fmt = av_guess_format("mpeg", NULL, NULL);
     }
     if (!fmt) {
 	G_warning(_("Unable to find suitable output format"));
@@ -312,7 +312,7 @@ int gsd_init_mpeg(const char *filename)
     }
 
     /* allocate the output media context */
-    oc = av_alloc_format_context();
+    oc = av_malloc(sizeof(AVFormatContext));
     if (!oc) {
 	G_warning(_("Out of memory"));
 	return (-1);
